OpenVINO Add precision type to inputs and int4 support

Vladimir9812 opened the pull request:
Hi, @lutzroeder, in this PR has been added support for int4 and uint4 layer precisions and have been enabled precisions for layer inputs

lutzroeder commented on source/openvino.js:
```diff
@@ -126,7 +126,16 @@ openvino.Graph = class {
+            const inputs = layer.inputs.map((input) => {
+                const to = layer.id + ':' + input.id;
+                const [outputLayerId, outputId] = net.edges[to].split(':');
+                const outputLayer = net.layers.find((layer) => layer.id === outputLayerId);
```
Lookup map and `find` can be inefficient.

Vladimir9812 replied:
What lookup map can you suggest?

lutzroeder replied:
`const layers = new Map(net.layers.map((entry) => [ entry.id, entry ]))` and `layers.get(outputLayerId))`.

lutzroeder commented on source/openvino.js:
```diff
@@ -126,7 +126,16 @@ openvino.Graph = class {
+                const outputLayer = net.layers.find((layer) => layer.id === outputLayerId);
+                if (outputLayer) {
+                    const output = outputLayer.outputs.find((output) => output.id === outputId);
+                    input.precision = output.precision;
```
* What exactly is this code doing?
* Can the `precision` of the output always assumed to be the same as the input.
* `input` contains source data that should be copied instead of modified.

Vladimir9812 replied:
Initially in XML file inputs don't have precision property, unlike outputs. So, I try to match inputs and outputs according to the edges map and assign precisions to inputs.
Yes, precision to the output always assumed to be the same as the input.

lutzroeder replied:
@Vladimir9812 "assumed" to be the same as in this code makes this assumption but it could be wrong or is this always the case for all OpenVINO models?

Vladimir9812 replied:
@lutzroeder, sorry for my long answers...Yes, it is true for all OpenVINO models.
Thank you for merge. Can I find a smaller int4 model and attach it to this PR?

lutzroeder replied:
Yes, that would be great.

Vladimir9812 replied:
[simple_model_int4.zip](https://github.com/lutzroeder/netron/files/6278507/simple_model_int4.zip)

Vladimir9812 commented on source/openvino.js:
```diff
@@ -126,7 +126,16 @@ openvino.Graph = class {
         for (const layer of this._const(net.layers, net.edges)) {
-            const inputs = layer.inputs.map((input) => this._argument(layer.id, layer.precision, input, net.edges));
+            const inputs = layer.inputs.map((input) => {
+                const to = layer.id + ':' + input.id;
```
Thank you, @lutzroeder, I will fix all your comments tomorrow during the day

Vladimir9812 commented on source/openvino.js:
```diff
@@ -936,21 +945,21 @@ openvino.TensorType = class {
             case 'f32':     this._dataType = 'float32'; break;
             case 'fp32':    this._dataType = 'float32'; break;
             case 'bf16':    this._dataType = 'bfloat16'; break;
+            case 'i4':      this._dataType = 'int4'; break;
```
This is a model with int4 for test
[int4-model.zip](https://github.com/lutzroeder/netron/files/6259225/int4-model.zip)

lutzroeder replied:
Can you provide a smaller model? 10 MB is a lot of a test file to test a data type.

Vladimir9812 commented:
@lutzroeder, I don't have any information about correctly decoding int4 dataType. So I've added a stub so far, but it is a temporary decision. I corrected all the other comments and PR is waiting for a second review round
